Issue 1 create navbar

Jared-Krajewski opened the pull request:
## This PR:

Resolves #1 , Colors are not Final so ignore them but good enough for merge. There are a few other small changes in here as well that were discussed in discord. (gradient for hero readability, capitalized a "w"  and started theme to hook up dark mode button)

---

## Screenshots 


https://github.com/codeforpdx/codepdx_website/assets/114430929/96c75dfd-09a2-45e1-a5ed-aad3647a0af0



---

## Future Steps/PRs Needed to Finish This Work (optional):

The logo/tag thing might change in the future. need to discuss at next meeting

Jared-Krajewski committed:
footprint of nav with darkmode implementation

Jared-Krajewski committed:
hooked up dark mode to custom theme

Jared-Krajewski committed:
added dark mode theme

Jared-Krajewski committed:
Merge remote-tracking branch 'origin/main' into issue-1-create-navbar

Jared-Krajewski committed:
logo added

Jared-Krajewski committed:
Merge commit '04edc4a9f149d50fc2c5721130556dbca14decd1' into issue-1-create-navbar

Jared-Krajewski committed:
placed top left content of nav bar

Jared-Krajewski committed:
added link styling

Jared-Krajewski committed:
more flex styling and visiblity change to hero

Jared-Krajewski committed:
added hamburger menu and popover

Jared-Krajewski committed:
fixed rose behind blob

andycwilliams commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+  right: 0,
+  top: -50,
+  height: '250px',
+  backgroundImage: { md: 'url(/assets/logoBlobNav.svg)', sm: 'none' },
```
This doesn't match the image name. Solved by either renaming the image itself to `logoBlobNav` or this line to `logoBlobnav`

Jared-Krajewski replied:
Im not sure what you are referring to. It is connected to the intended svg on line 24. Any name change will break that.

Jared-Krajewski replied:
the icon and brand name both redirect home, It's pretty commonplace to do that and not leave a home link. It only appears in the dropdown because the brand name drops off.  No plan on putting a home tab, might even remove it from dropdown in the future.

andycwilliams replied:
In my assets folder, the image is titled `logoBlobnav.svg`. It's also what shows when I look through the `Files changed` tabbed. One has a capitalized 'Nav` while the other doesn't.

Jared-Krajewski replied:
![Screenshot 2023-10-26 at 6 00 45 PM](https://github.com/codeforpdx/codepdx_website/assets/114430929/9191a306-540c-495f-a9cd-990c6695a0fe)
Got it, thats a glitch of some sort, its capitalized on my local.  I fixed it on GH.

Jared-Krajewski replied:
The button does an eclipse transition when clicked and wont harm anyone if they tap it and its effect will be obvious. That is the icon MUI offers for dark mode (only one I saw). for the sake of trying to get the site up I'm OK with using it as is.

andycwilliams replied:
In that case this PR looks good enough to me

andycwilliams commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+        <Box sx={logoBlobStyle}></Box>
+        <Link to="/" style={textStyle} aria-label="Home">
+          <Box
+            component="img"
```
Do you plan on abstracting more styling as we have elsewhere? No major issue, just wondering.

Jared-Krajewski replied:
a fair chunk of it is just unique enough where it wont save any code. Im pretty sure abstracting and setting up a function to map the 2 links would be more code. some of it has to be done at the component level as well.

andycwilliams commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+          </IconButton>
+        </Hidden>
+        <IconButton edge="end" color="inherit" onClick={handleThemeChange}>
+          {darkMode ? <Brightness7 /> : <Brightness4 />}
```
For the dark mode button, maybe add it to the menu for smaller viewports? It might be the least interacted-with button on there. A lot of sites that I've seen place it inside a menu for all viewports.

Jared-Krajewski replied:
Once I make sure the setup is right It wont exist on mobile.  It will just connect with mobile theme or display brightness. That will come once theme updates are done.

andycwilliams reviewed (commented):
Looking very nice. Dark mode looks kind of cool with the logo.

Do we want a `Home` tab along with `Projects` and `Volunteer`? Clicking the logo does that but I think sometimes people don't think of that, so having a named tab would help. It would also make navigating through each page quicker if they're close together. Plus there's already a `Home` option in the menu for the mobile version.

Also maybe label the button? It may not be immediately obvious what that icon does. I've never actually seen one like it, personally. So I wonder if it comes off as more of a second logo than a dark mode option.

All in all, this should be very close to ready.

Jared-Krajewski committed:
Rename logoBlobnav.svg to logoBlobNav.svg

fixed spelling in GH since pushing the name change does nothing.

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+  Toolbar,
+  Typography
+} from '@mui/material';
+import MenuIcon from '@mui/icons-material/Menu';
```
Can combine the icon imports to same line if want them like line 13, or split them to three lines if want them like line 12...

Docs say to make like line 12 for bundle size [here](https://mui.com/material-ui/icons/)

Jared-Krajewski replied:
good to know, ill do them like line 12, also surprised i didnt notice and put them all into line 13.

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+  Typography
+} from '@mui/material';
+import MenuIcon from '@mui/icons-material/Menu';
+import { Brightness4, Brightness7 } from '@mui/icons-material';
```
These for light/dark mode I think are perfectly fine, my personal vote would be the "dark mode" icon and the "light mode" icon though. Not gonna fight for it, as what was chosen is not an issue at all.

Jared-Krajewski replied:
I just tweaked them, Oddly before reading your comment but playing with it while putting the new imports I thought the same thing.

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+} from '@mui/material';
+import MenuIcon from '@mui/icons-material/Menu';
+import { Brightness4, Brightness7 } from '@mui/icons-material';
+import { BrowserRouter as Route, Link } from 'react-router-dom';
```
This `BrowserRouter as Route` can be deleted, not being used.

Jared-Krajewski replied:
maybe that auto import extension is not as great as I thought out of the box.

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+  backgroundImage: { md: 'url(/assets/logoBlobNav.svg)', sm: 'none' },
+  backgroundRepeat: 'no-repeat',
+  display: {
+    lg: 'flex',
```
this line (27) makes the blob show from `lg` and up (1200px), but styling on line 24 seems like you want it from `md` and up (900px).   The 900px mark makes more sense to me & I tested for it.  I would switch this line to `md: 'flex'`

Jared-Krajewski replied:
That line didn't do anything anymore, forgot to delete 
```
  display: {
    lg: 'flex',
    xs: 'none'
  }
 ```

xscottxbrownx replied:
Oh right... so let's clean it up a little more then.  How about instead of rendering a `backgroundImage` based on screen size - we render the `Box` holding it that way?   Which I just noticed could be self-closing too.

```js
backgroundImage: 'url(/assets/logoBlobNav.svg)',

<Box sx={logoBlobStyle} display={{ xs: 'none', md: 'block' }} />
```

Jared-Krajewski replied:
changed

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+  // anchor for menu to pop from its icon
+  const anchorRef = useRef(null);
+
+  const handleThemeChange = () => {
```
### lines 45-51:

Do you know what best practice is for these state changes? Just for my knowledge and learning.   

I've done them both ways, but don't really know what the difference is...

Like when to just use just bang operator
&
when to pass in a function that takes current state and bang operator on that current state
??

Jared-Krajewski replied:
To the best of my knowledge react has batch state updates and using  the format ` setMenuOpen(!menuOpen)` could cause some issues if its spammed, this is more of an issue if multiple async values can come in and change state. Not a big deal since its only light/dark mode so either would work here.

xscottxbrownx replied:
> To the best of my knowledge react has batch state updates and using the format ` setMenuOpen(!menuOpen)` could cause some issues if its spammed, this is more of an issue if multiple async values can come in and change state. Not a big deal since its only light/dark mode so either would work here.

ahh yes, gotcha.

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+            src={'/assets/rose_logo.png'}
+            sx={{
+              width: '75px',
+              mr: '30px',
```
Think the margin right should be less when blob disappears, seems like the logo and title are disconnected & I feel they kind of work together. I address this in my sample code for lines 75-92.

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+            sx={{
+              width: '75px',
+              mr: '30px',
+              ml: { xs: '0px', sm: '30px', lg: '20px' }
```
why more margin on small screen vs large? maybe makes more sense for `sm: '10px'`?

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+    <AppBar position="static" color={darkMode ? 'dark' : 'primary'}>
+      <Toolbar sx={{ height: '100px', alignItems: 'center' }}>
+        <Box sx={logoBlobStyle}></Box>
+        <Link to="/" style={textStyle} aria-label="Home">
```
### the logo and `CODE PDX` text...

Test this for lines 75-92:

```js
<Link to="/" style={textStyle} aria-label="Home">
    <Box
        component="img"
        position="relative"
        alt="CODE PDX logo"
        src={'/assets/rose_logo.png'}
        sx={{
            width: '75px',
            mr: { xs: '0px', md: '30px' },
            ml: { xs: '0px', md: '20px' }
        }}
     />
</Link>
<Typography
    variant="h4"
    component="h1"
    display={{ xs: 'none', sm: 'block' }}
    pl={{ xs: '10px', md: '15px' }}
>
    <Link to="/" style={textStyle} aria-label="Home">
        CODE PDX
    </Link>
</Typography>
```
Hard to explain in words, but I think it makes for smoother transitions with better spacing on the breakpoints.

Jared-Krajewski replied:
changed

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+            }}
+          />
+        </Link>
+        <Typography variant="h4" display={{ xs: 'none', sm: 'block' }} pl={'15px'}>
```
This `Typography` should have `component=h1` in my opinion for semantics.  As it describes what the page/site is, and all the section headers we are setting up as `h2`s.  

Not sure what to do for semantics when this disappears yet.

Also, I think `pl` should be smaller when blob disappears (see other comments and example code.)

Jared-Krajewski replied:
changed to 10px. I feel like thats about as close as it can get.

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+  );
+
+  return (
+    <AppBar position="static" color={darkMode ? 'dark' : 'primary'}>
```
Not sure exactly what `AppBar` and `Toolbar` become, but we need to get a `<nav>` in this component somewhere (or `component="nav"`.)

---

Look at the "Sectioning Content Examples" just under [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#labeling_section_content) linked section.  Since we have some nav links in footer as well, I think this is an example of exactly what to do for accessibility stuff (so screen reader/etc knows what is the primary navigation and secondary / less important navigation.)

Jared-Krajewski replied:
forgot to add `component='nav'` good catch. fixed

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+        {/* Hamburger menu for smaller viewports */}
+        <Hidden smUp>
+          <IconButton
+            sx={{ ml: 'auto' }}
```
change the styling of this `IconButton` to:
`sx={{ ml: 'auto', mr: 1 }}`

adds some space between menu icon and dark/light icon. Feel it looks better and will probably be less frustrating to tap on mobile.

Jared-Krajewski replied:
agree, changed.

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,147 @@
+        <IconButton edge="end" color="inherit" onClick={handleThemeChange}>
+          {darkMode ? <Brightness7 /> : <Brightness4 />}
+        </IconButton>
+        <Popover
```
I'm sure it might be annoying to change...

BUT I don't know how easy it will be to navigate in this type of `Popover` dropdown-like menu on mobile devices. Looks real small.

Might want to think of changing it to something larger, like maybe this [drawer](https://mui.com/material-ui/react-app-bar/#responsive-app-bar-with-drawer) ?

xscottxbrownx replied:
Oh, and if not changing - I think changing link color (or even the menu box color maybe) to more of our theming would be good.  The links currently show as the generic browser link colors for me.

Jared-Krajewski replied:
had some strong opinions about how the `Drawer` looked so went for `Popover`. fixing links, making it wider, and centering text (while fixing the style of text).

xscottxbrownx reviewed (changes requested):
### I have a good amount...

You can decide what you want to implement, what you want to create an issue for and do later, or what you want to ignore/discuss.

----

I also was going to comment about the black text on Hero section button (because others are white), but realized the Figma shows black. I missed that on volunteer brief section of homepage.

Jared-Krajewski committed:
some fixes from PR

Jared-Krajewski committed:
fixed all PR issues except popover

Jared-Krajewski committed:
merge main

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,157 @@
+          </IconButton>
+        </Hidden>
+        <IconButton edge="end" color="inherit" onClick={handleThemeChange}>
+          {darkMode ? <Brightness3Icon /> : <Brightness7Icon />}
```
flip these two icons... My opinion, but I feel when in light mode the icon should be the moon, and in dark mode the icon should be the sun.

It's fine if you disagree, not an issue - just how I think personally.

Jared-Krajewski replied:
fixed with a better implementation (toggle)

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,157 @@
+        </IconButton>
+        <Popover
+          open={menuOpen}
+          anchorEl={anchorRef.current}
```
as discussed already, if you can make the anchor point possibly the bottom of the navbar, so it won't cover it at all (and won't cause an issue if resize screen while menu is open - icon disappears so menu covers entire navbar.)

Jared-Krajewski replied:
anchored off bottom of window

xscottxbrownx commented on src/components/global/Navbar.jsx:
```diff
@@ -1,8 +1,157 @@
+
+  const menuItems = (
+    <List>
+      <ListItem sx={menuTextStyle} component={Link} to="/" onClick={handleClose}>
```
Maybe some styling on hover?   Either change link color, or maybe even the background color of that link section in menu?

Actually probably do this for all the links in the navbar, not just the mobile menu (like we have in footer.) Better indicator than solely the pointer cursor.

Jared-Krajewski replied:
added

xscottxbrownx reviewed (changes requested):
Couple more, most we have already discussed.

Including the svg name again - stops working due to different casing.

Jared-Krajewski committed:
fixed dropdown

Jared-Krajewski committed:
added hover effect for nav links

Jared-Krajewski committed:
fixed imports

Jared-Krajewski committed:
Rename logoBlobnav.svg to logoBlobNav.svg

Github keeps forcing lowercase even tho its caps on local, despite all efforts. forcing back to caps.

Jared-Krajewski committed:
Update App.jsx

removed extra spaces
